python/tests/variable_creation_test.py

```diff
@@ -230,11 +230,11 @@ def test_array_creates_correct_variable():
     assert sc.identical(var, expected)
 
 
-def test_array_needs_nonempty_dims():
-    with pytest.raises(ValueError):
-        sc.array(dims=[], values=[])
-    with pytest.raises(ValueError):
-        sc.array(dims=None, values=[])
+def test_array_empty_dims():
+    assert sc.identical(sc.array(dims=[], values=[1]), sc.scalar([1]))
+    a = np.asarray(1.1)
+    assert sc.identical(sc.array(dims=None, values=a), sc.scalar(1.1))
+    assert sc.identical(sc.array(dims=[], values=a), sc.scalar(1.1))
```

Comment:
Seems logical. If the dimensions are empty and you only have a single value in your values.

Reply:
Did you check this also works if single value for variances is supplied?

Reply:
`sc.array` and `sc.scalar` both use `sc.Variable` internally so this should be covered by those tests, since that aspect is orthogonal to what we check here, I believe.